src/nccl_impl.cpp

```diff
@@ -57,8 +44,18 @@ void NCCLCommunicator::nccl_setup() {
   AL_CHECK_NCCL(ncclCommInitRank(&m_nccl_comm, size(), nccl_id, rank()));
 }
 
-void NCCLCommunicator::nccl_destroy() {
-  AL_CHECK_NCCL(ncclCommDestroy(m_nccl_comm));
+NCCLCommunicator::~NCCLCommunicator() {
```

Comment:
```suggestion
NCCLCommunicator::~NCCLCommunicator() noexcept {
```
It's now truly `noexcept` since the exception does not leave the function.